modules/dynamic-provisioning-azure-file-definition.adoc

```diff
@@ -0,0 +1,59 @@
+// Module included in the following assemblies
+//
+// * storage/dynamic-provisioning.adoc
+
+[id="azure-file-definition_{context}"]
+= Azure File object definition
+
+The Azure File StorageClass uses secrets to store the Azure storage account name
+and the storage account key required to create an Azure Files share. These
+permissions are created as part of the following procedure.
+
+.Procedure
+
+. Define a ClusterRole that allows access to create and view secrets.
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+#  name: system:azure-cloud-provider
+  name: persistent-volume-binder-role
+rules:
+- apiGroups: ['']
+  resources: ['secrets']
+  verbs:     ['get','create']
+----
+
+. Add the ClusterRole to the ServiceAccount.
++
+----
+$ oc adm policy add-cluster-role-to-user persistent-volume-binder-role system:serviceaccount:kube-system:persistent-volume-binder
+----
+
+. Create the Azure File StorageClass.
++
+[source,yaml]
+----
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: <azure-file> <1>
+provisioner: kubernetes.io/azure-file
+parameters:
+  location: eastus <2>
+  skuName: Standard_LRS <3>
+  storageAccount: <storage-account> <4>
+reclaimPolicy: Delete
+volumeBindingMode: Immediate
+----
+<1> Name of the StorageClass. This will be used by the PersistentVolumeClaim
```

Comment:
s/This will be used by the PersistentVolumeClaim
to determine the provisioning StorageClass./
The PersistentVolumeClaim uses this name to determine the StorageClass to provision. (Or "that it provisions" - "provisioning" is ambiguous.)

Reply:
I've reworked this to be clearer.